core/src/autogluon/core/searcher/bayesopt/autogluon/gp_fifo_searcher.py

```diff
@@ -200,18 +165,116 @@ def update(self, config: Candidate, reward: float, **kwargs):
         :param config:
         :param reward:
         """
+        config_ext = self._config_ext_update(config, **kwargs)
         crit_val = self.map_reward(reward)
         metrics = dictionarize_objective(crit_val)
         if 'elapsed_time' in kwargs:
             metrics[self.cost_metric_name] = kwargs['elapsed_time']
         self.state_transformer.label_candidate(CandidateEvaluation(
-            candidate=copy.deepcopy(config), metrics=metrics))
+            candidate=config_ext, metrics=metrics))
         if self.debug_log is not None:
-            config_id = self.debug_log.config_id(config)
+            config_id = self.debug_log.config_id(config_ext)
             msg = "Update for config_id {}: reward = {}, crit_val = {}".format(
                 config_id, reward, crit_val)
             logger.info(msg)
 
+    def dataset_size(self):
+        return len(self.state_transformer.state.candidate_evaluations)
+
+    def get_params(self):
+        """
+        Note: Once MCMC is supported, this method will have to be refactored.
+
+        :return: Dictionary with current hyperparameter values
+        """
+        return self.state_transformer.get_params()
+
+    def set_params(self, param_dict):
+        self.state_transformer.set_params(param_dict)
+
+    def get_state(self):
+        """
+        The mutable state consists of the GP model parameters, the
+        TuningJobState, and the skip_optimization predicate (which can have a
+        mutable state).
+        We assume that skip_optimization can be pickled.
+
+        """
+        state = {
+            'model_params': self.get_params(),
+            'state': encode_state(self.state_transformer.state),
+            'skip_optimization': self.state_transformer.skip_optimization,
+            'random_state': self.random_state}
+        if self.debug_log is not None:
+            state['debug_log'] = self.debug_log.get_mutable_state()
+        return state
+
+
+class GPFIFOSearcher(GPSearcher):
+    """
+    Supports standard GP-based hyperparameter optimization, when used with a
+    FIFO scheduler.
+
+    """
+    def __init__(
+            self, hp_ranges: HyperparameterRanges, random_seed: int,
+            gpmodel: GPModel, model_args: GPModelArgs,
+            map_reward: MapReward,
+            acquisition_class: Type[AcquisitionFunction],
+            init_state: TuningJobState = None,
+            local_minimizer_class: Type[LocalOptimizer] = DEFAULT_LOCAL_OPTIMIZER_CLASS,
+            skip_optimization: SkipOptimizationPredicate = None,
+            num_initial_candidates: int = DEFAULT_NUM_INITIAL_CANDIDATES,
+            num_initial_random_choices: int = DEFAULT_NUM_INITIAL_RANDOM_EVALUATIONS,
+            initial_scoring: Optional[str] = None,
+            first_is_default: bool = True,
+            debug_log: Optional[DebugLogPrinter] = None,
+            cost_metric_name: Optional[str] = None,
```

Comment:
cost_metric_name is missing a :param description below.

Reply:
OK